core/state-machine/src/ext.rs

```diff
@@ -264,6 +264,21 @@ where
 		});
 	}
 
+	fn clear_child_prefix(&mut self, storage_key: &[u8], prefix: &[u8]) {
+		let _guard = panic_handler::AbortGuard::new(true);
+		if is_child_storage_key(prefix) {
```

Comment:
I don't quite understand this `if`. `prefix` is a key in child storage, so it has nothing to do with root's `is_child_storage_key`?

Reply:
I should remove this (all new child function) from the PR, it is not needed (I was making the implementation at somepoint because I wanted to generalize child trie logic to root trie to, but that is an open question (making root trie a child trie with parent trie &[]) -> this result to hierarchical trie addressing and is out of this pr scope.